src/mozilla_sec_eia/models/sec_eia_record_linkage/transform_sec_input.py

```diff
@@ -0,0 +1,392 @@
+
+@asset(
+    ins={
+        "basic_10k_dfs": AssetIn("basic_10k_company_info"),
+        "sec10k_filing_metadata_dfs": AssetIn("sec10k_filing_metadata"),
+    },
+)
+def core_sec_10k__filers(
+    basic_10k_dfs: dict[str, pd.DataFrame],
+    sec10k_filing_metadata_dfs: dict[str, pd.DataFrame],
+) -> pd.DataFrame:
+    """Asset for creating a cleaned basic 10k table with EIA utility matched.
+
+    Flatten the table across time to only keep the most recent record
+    for each unique company name and address pair. Clean table and link filers
+    to EIA utilities.
+    """
+    basic_10k_df = pd.concat(basic_10k_dfs.values())
+    sec10k_filing_metadata = pd.concat(sec10k_filing_metadata_dfs.values())
+    basic_10k_df = transform_basic10k_table(basic_10k_df, sec10k_filing_metadata)
+    out_df = basic_10k_df.fillna(np.nan).reset_index(names="record_id")
+    # match EIA utilities to filers
+    # TODO: Here we conduct the match to EIA and add on a column with utility_id_eia
+    return out_df
+
+
+@asset(
+    ins={
+        "sec_10k_filers_matched_df": AssetIn("core_sec_10k__filers"),
+        "clean_ex21_df": AssetIn("transformed_ex21_subsidiary_table"),
+        "clean_eia_df": AssetIn("core_eia__parents_and_subsidiaries"),
+    },
+)
+def out_sec_10k__parents_and_subsidiaries(
+    sec_10k_filers_matched_df: pd.DataFrame,
+    clean_ex21_df: pd.DataFrame,
+    clean_eia_df: pd.DataFrame,
+) -> pd.DataFrame:
+    """Asset for creating an SEC 10K output table.
+
+    Add in Ex. 21 subsidiaries and link them to already present
+    filing companies. Create an sec_company_id for subsidiaries
+    that aren't linked to a CIK.
+    """
+    ex21_df_with_cik = match_ex21_subsidiaries_to_filer_company(
+        basic10k_df=sec_10k_filers_matched_df, ex21_df=clean_ex21_df
+    )
+    sec_10k_filers_matched_df = sec_10k_filers_matched_df.merge(
+        ex21_df_with_cik[["central_index_key", "parent_company_cik", "own_per"]],
+        how="left",
+        on="central_index_key",
+    )
+    # get the subsidiary companies that weren't matched to a 10K filing company
+    ex21_non_filing_subs_df = ex21_df_with_cik[
+        ex21_df_with_cik["central_index_key"].isnull()
+    ]
+    ex21_non_filing_subs_df.loc[:, "files_10k"] = False
+    # the last step is to take the EIA utilities that haven't been matched
+    # to a filer company, and merge them by company name onto the Ex. 21 subs
+    unmatched_eia_df = clean_eia_df[
+        ~clean_eia_df["utility_id_eia"].isin(
+            sec_10k_filers_matched_df.utility_id_eia.unique()
+        )
+    ].drop_duplicates(subset="company_name")
+    ex21_non_filing_subs_df = ex21_non_filing_subs_df.merge(
+        unmatched_eia_df[["utility_id_eia", "company_name"]],
+        how="left",
+        on="company_name",
+    ).drop_duplicates(subset="sec_company_id")
+    logger.info(
+        f"Ex. 21 subsidiary names matched to an EIA utility name: {len(ex21_non_filing_subs_df["utility_id_eia"].unique())}"
+    )
```

Comment:
we should probably have more logging throughout these transformations and record them with mlflow